src/Dfe.PlanTech.Web/Controllers/CheckAnswersController.cs

```diff
@@ -15,26 +16,79 @@ public class CheckAnswersController : BaseController<CheckAnswersController>
 {
     public const string PAGE_SLUG = "check-answers";
 
-    public CheckAnswersController(ILogger<CheckAnswersController> logger) : base(logger)
+    public const string INLINE_RECOMMENDATION_UNAVAILABLE_ERROR_MESSAGE =
+        "Unable to determine your recommendation. Please try again.";
+
+    private readonly IUser _user;
+    private readonly IGetSectionQuery _getSectionQuery;
+    private readonly IProcessCheckAnswerDtoCommand _processCheckAnswerDtoCommand;
+    private readonly IGetPageQuery _getPageQuery;
+    private readonly ICalculateMaturityCommand _calculateMaturityCommand;
+
+    public CheckAnswersController(IUser user, IGetSectionQuery getSectionQuery,
+        IProcessCheckAnswerDtoCommand processCheckAnswerDtoCommand, IGetPageQuery getPageQuery, ICalculateMaturityCommand calculateMaturityCommand,
+        ILogger<CheckAnswersController> logger) : base(logger)
     {
+        _user = user;
+        _getSectionQuery = getSectionQuery;
+        _processCheckAnswerDtoCommand = processCheckAnswerDtoCommand;
+        _getPageQuery = getPageQuery;
+        _calculateMaturityCommand = calculateMaturityCommand;
     }
 
     [HttpGet("{sectionSlug}/check-answers")]
-    public async Task<IActionResult> CheckAnswersPage(string sectionSlug, [FromServices] IUser user, [FromServices] IGetSectionQuery getSectionQuery, [FromServices] IProcessCheckAnswerDtoCommand processCheckAnswerDtoCommand, [FromServices] IGetPageQuery getPageQuery, CancellationToken cancellationToken = default)
+    public async Task<IActionResult> CheckAnswersPage(string sectionSlug, CancellationToken cancellationToken = default)
     {
         if (string.IsNullOrEmpty(sectionSlug)) throw new ArgumentNullException(nameof(sectionSlug));
 
-        var section = await getSectionQuery.GetSectionBySlug(sectionSlug, cancellationToken) ??
-                            throw new KeyNotFoundException($"Could not find section for {sectionSlug}");
+        var checkAnswersViewModel = await GenerateCheckAnswersViewModel(sectionSlug, null, cancellationToken);
+
+        if (checkAnswersViewModel == null) return this.RedirectToSelfAssessment();
+
+        return View("CheckAnswers", checkAnswersViewModel);
+    }
+
+    [HttpPost("ConfirmCheckAnswers")]
+    public async Task<IActionResult> ConfirmCheckAnswers(string sectionSlug, int submissionId, string sectionName,
```

Comment:
By returning a `View` here, you'll end up with the URL being this 

<img width="877" alt="image" src="https://github.com/DFE-Digital/plan-technology-for-your-school/assets/131891854/673a8492-7e34-4a6f-9877-496b69863922">

IMO it'll be better to simply redirect to the `CheckAnswersPage` action instead to prevent that